src/main/java/gregtech/api/recipes/recipeproperties/DimensionProperty.java

```diff
@@ -0,0 +1,82 @@
+package gregtech.api.recipes.recipeproperties;
+
+import gregtech.api.worldgen.config.WorldGenRegistry;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.resources.I18n;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntList;
+
+public class DimensionProperty extends RecipeProperty<DimensionProperty.DimensionPropertyList> {
+
+    public static final String KEY = "dimension";
+
+    private static DimensionProperty INSTANCE;
+
+    private DimensionProperty() {
+        super(KEY, DimensionPropertyList.class);
+    }
+
+    public static DimensionProperty getInstance() {
+        if (INSTANCE == null)
+            INSTANCE = new DimensionProperty();
+        return INSTANCE;
+    }
+
+    @Override
+    public void drawInfo(Minecraft minecraft, int x, int y, int color, Object value) {
+        DimensionPropertyList list = castValue(value);
+
+        if (list.whiteListDimensions.size() > 0)
+            minecraft.fontRenderer.drawString(I18n.format("gregtech.recipe.dimensions",
+                    getDimensionsForRecipe(castValue(value).whiteListDimensions)), x, y, color);
+        if (list.blackListDimensions.size() > 0)
+            minecraft.fontRenderer.drawString(I18n.format("gregtech.recipe.dimensions_blocked",
+                    getDimensionsForRecipe(castValue(value).blackListDimensions)), x, y, color);
+    }
+
+    private static String getDimensionsForRecipe(IntList value) {
+        Int2ObjectMap<String> dimNames = WorldGenRegistry.getNamedDimensions();
+        StringBuilder builder = new StringBuilder();
+        for (int i = 0; i < value.size(); i++) {
+            builder.append(dimNames.getOrDefault(value.getInt(i), String.valueOf(value.getInt(i))));
+            if (i != value.size() - 1)
+                builder.append(", ");
+        }
+        String str = builder.toString();
+
+        if (str.length() >= 13) {
+            str = str.substring(0, 10) + "..";
+        }
+        return str;
+    }
+
+    // It would've been better to have one list and swap between blacklist and whitelist, but that would've been
+    // a bit awkward to apply to the property in practice.
+    public static class DimensionPropertyList {
+
+        public static DimensionPropertyList EMPTY_LIST = new DimensionPropertyList();
+
+        public IntList whiteListDimensions = new IntArrayList();
+        public IntList blackListDimensions = new IntArrayList();
+
+        public void add(int key, boolean toBlacklist) {
+            if (toBlacklist) blackListDimensions.add(key);
+            else whiteListDimensions.add(key);
```

Comment:
If someone whitelists the overworld and then later blacklists it in a recipebuilder call, we should probably handle this by removing it from the other list (so that the last call gets priority). This is a weird case but still would be best to be stronger than purely giving whitelist priority as is the case in `checkDimension`